doc/lua_api.txt

```diff
@@ -7885,6 +7885,42 @@ Used by `minetest.register_node`.
         -- Default: nil
         -- Warning: making a liquid node 'floodable' will cause problems.
 
+        on_fall = function(pos, node)
+        -- Called before node starts to fall.
+        -- Optionally should return `<boolean>, <obj>`.
+        -- If `boolean` is returned as a boolean, the builtin falling node
+        -- handling is skipped.
+        -- `boolean=true` means an object was created by the function which
+        -- must be returned in `obj`.
+        -- `boolean=false` means no object was created and `obj` must be `nil`;
+        -- `boolean=nil` means the builtin falling handling is executed.
+        -- The builtin falling node falling means that the node at `pos` is
+        -- removed and a `__builtin:falling_node` entity spawns in its place.
+        -- `after_fall` is only called if `boolean` `nil`.
+        -- default: nil
```

Comment:
Now the api states that the `on_fall` callback must always add a falling node entity to allow to propagate the falling check. Ie. it's impossible to spawn a totally different object.
This makes returning something else than `nil` pretty useless.

Also, if the returned bool is `false`, the node must not have been removed, else we have floating nodes, the doc is missing this information. (I'm referring to [this](https://github.com/minetest/minetest/blob/c6e03b5ba91c6578f16f7ee2b6818cea4aa70992/builtin/game/falling.lua#L596).)

I think anything that doesn't involve forcing to return a `__builtin:falling_node` entity will probably come with an api break of `minetest.spawn_falling_node()`.
Possible options include:
- 1. Allow `minetest.spawn_falling_node()` to return `nil` as 2nd return value.
- 2. Make `minetest.spawn_falling_node()` return `false` if `on_fall` has returned a boolean. Maybe return `false, bool`.
- 3. Make `minetest.spawn_falling_node()` return `false` and do nothing if `def.on_fall` exists. And add a 2nd function that also calls `on_fall`, naming suggestions follow:
  * 1. Add a function version number: `minetest.spawn_falling_node2()`
  * 2. Use a super descriptive and long name: `minetest.spawn_falling_node_and_also_call_on_fall_in_node_definition()`
  * 3. Reorder the words, à la `.node_dig()` and `.dig_node()`: `minetest.spawn_node_falling()`
  * 4. Add spelling errors: `minetest.spawn_faling_node()`